plugins/python/python_plugin.c

```diff
@@ -1297,14 +1298,41 @@ void uwsgi_python_init_apps() {
 
 }
 
-void uwsgi_python_master_fixup(int step) {
 
+void uwsgi_python_pre_uwsgi_fork() {
+	// Replicate what os.fork() by acquiring import lock as well as acquiring gil before forking
+	uwsgi_log("UWSGI_PYTHON_PRE_UWSGI_FORK\n");
+	UWSGI_GET_GIL
+	_PyImport_AcquireLock();
+}
+
+
+void uwsgi_python_post_uwsgi_fork(int step) {
+	if (!uwsgi.master_process) return;
+
+	if (uwsgi.has_threads) {
+		uwsgi_log("UWSGI_PYTHON_POST_UWSGI_FORK %d\n", step);
+		if (step == 0) {
+			// release locks in master process
+			_PyImport_ReleaseLock();
+			UWSGI_RELEASE_GIL
+		}
+		else {
+			// ensure thread state and locks is cleaned in child process
+			PyOS_AfterFork();
```

Comment:
Is `PyOS_AfterFork()` being called elsewhere?

Reply:
I removed the other two places it was used.